Automattic/wp-calypso
# WPCOM: Remove `undocumented().me()` completely

#### Changes proposed in this Pull Request

This PR migrates the last `wpcom.undocumented()` method - `getReceipt()` to `wpcom.req`. 

We also remove the unnecessary `undocumented().me()`, while removing its last usages, that were just proxying requests to the `wpcom.js` library.

Part of the ongoing effort to get rid of `wpcom.undocumented()` and part of #58458.

~Depends on #58452 and has to be rebased once it gets merged.~

#### Testing instructions

* Purchase a plan and verify the thank you page still works well and displays purchase details correctly.
* Navigate to `/me/purchases/billing` and click on a receipt. Verify the page still works well.
* Verify you can still change and save settings in Me -> Account Settings correctly.
* Verify you can still change your 2FA SMS settings (namely, the phone number).
* Verify you can still change your email address through the domain contact info management page.
* Verify all tests still pass.
